ReactiveXComponent/XCApi/XComponentApi.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using ReactiveXComponent.Parser;
+using ReactiveXComponent.RabbitMQ;
+
+namespace ReactiveXComponent.XCApi
+{
+    public class XComponentApi : IXComponentApi
+    {
+        private bool _disposed = false;
+        
+        private readonly XCConnection _xcConnection;
+        private readonly XCSessionFactory _xcSessionFactory;
+        private XCSession _xcSession;
+
+        public XComponentApi(IRabbitMqPublisherFactory publisherFactory, IRabbitMqConsumerFactory consumerFactory, IRabbitMqConnection connection, DeploymentParser parser)
+        {
+            _xcConnection = new XCConnection(connection);
```

Comment:
you never use your connection !! strange
your connection should create a session...